.dotnet/src/Custom/Chat/ChatMessage.cs

```diff
@@ -78,7 +78,7 @@ internal ChatMessage(ChatMessageRole role, IEnumerable<ChatMessageContentPart> c
         }
     }
 
-    internal ChatMessage(ChatMessageRole role, string content)
+    internal ChatMessage(ChatMessageRole role, string content = null)
```

Comment:
nit: The `content` is required in all cases except in `AssistantChatMessage` when there are either `ChatToolCalls` or a `FuctionCall`. Since this is the exception and not the rule, could you keep this `content` parameter as not defaulted and then explicitly pass `null` in those two cases? i.e., in the Custom/Chat/AssistantChatMessage.cs file:

```csharp
public AssistantChatMessage(IEnumerable<ChatToolCall> toolCalls)
    : base(ChatMessageRole.Assistant, null)
```

And also:

```csharp
public AssistantChatMessage(ChatFunctionCall functionCall)
    : base(ChatMessageRole.Assistant, null)
```

Reply:
>could you keep this content parameter as not defaulted and then explicitly pass null in those two cases? i.e., in the Custom/Chat/AssistantChatMessage.cs file:

If we don't default to null here, then how can I call `base(ChatMessageRole.Assistant, null)` from AssistantChatMessage ctor? do you want me to cast it?

```csharp
public AssistantChatMessage(IEnumerable<ChatToolCall> toolCalls)
     : base(ChatMessageRole.Assistant, (string)null)
```

Reply:
```csharp
internal ChatMessage(ChatMessageRole role, string content = null)
{
...
}
```
This is an internal constructor, so we can set the default value of `content` to `null` and call it in every case, regardless of whether `content` is required or not. This way, we avoid the need for casting.